src/Data/Array/ST.purs

```diff
@@ -83,6 +87,12 @@ thaw = copyImpl
 sort :: forall a h. Ord a => STArray h a -> ST h (STArray h a)
 sort = sortBy compare
 
+-- | Remove the first element from an array and return that element.
+shift :: forall h a. STArray h a -> ST h (Maybe a)
+shift = shiftImpl Just Nothing
+
+foreign import shiftImpl :: forall h a. (a -> Maybe a) -> Maybe a -> STArray h a -> ST h (Maybe a)
```

Comment:
Can we use rank-n types to make this a little safer please? If you write e.g. `(forall b. b -> Maybe b)` for the `Just` argument, then there are only two possibilities for the function which can go in there (i.e. either `const Nothing` or `Just`). This also makes it a little clearer how this is intended to be used. Also could you please put each argument on its own line to match the formatting for existing FFI imports in this file?